token/js/src/extensions/extensionType.ts

```diff
@@ -176,17 +185,32 @@ export function getAccountTypeOfMintType(e: ExtensionType): ExtensionType {
     }
 }
 
-function getLen(extensionTypes: ExtensionType[], baseSize: number): number {
+function getLen(
+    extensionTypes: ExtensionType[],
+    baseSize: number,
+    variableLengthExtensions?: [ExtensionType, number][]
+): number {
     if (extensionTypes.length === 0) {
         return baseSize;
     } else {
-        const accountLength =
+        let accountLength =
             ACCOUNT_SIZE +
             ACCOUNT_TYPE_SIZE +
             extensionTypes
                 .filter((element, i) => i === extensionTypes.indexOf(element))
                 .map((element) => addTypeAndLengthToLen(getTypeLen(element)))
                 .reduce((a, b) => a + b);
+        if (variableLengthExtensions) {
+            accountLength += variableLengthExtensions
+                .filter((element, i) => i === variableLengthExtensions.indexOf(element))
```

Comment:
I might be missing something, but I don't think this filter will do what it intends. It'll only catch if someone provides the exact same size and extension type, but won't de-dupe based on extension type.

How about passing in a dict or object as `[key: ExtensionType]: number` instead?

Reply:
Ah, clever! Type-safe against duplicates, too!